src/controllers/networks/networks.ts

```diff
@@ -68,69 +73,239 @@ export class NetworksController extends EventEmitter {
+            ...relayerNetwork
+          }
+          return
+        }
+
+        // If the network is custom we assume predefinedConfigVersion = 0
+        if (storedNetwork.predefinedConfigVersion === undefined) {
+          storedNetwork.predefinedConfigVersion = 0
+        }
+
+        // Mechanism to force an update network preferences if needed
+        const shouldOverrideStoredNetwork =
+          relayerNetwork.predefinedConfigVersion > 0 &&
+          relayerNetwork.predefinedConfigVersion > storedNetwork.predefinedConfigVersion
+
+        if (shouldOverrideStoredNetwork) {
+          updatedNetworks[chainId.toString()] = {
+            ...(predefinedNetworks.find((n) => n.chainId === relayerNetwork.chainId) || {}),
+            ...relayerNetwork,
+            id: storedNetwork?.id || relayerNetwork.id,
+            rpcUrls: [...new Set([...relayerNetwork.rpcUrls, ...storedNetwork.rpcUrls])]
+          }
+        } else {
+          updatedNetworks[chainId.toString()] = {
+            ...storedNetwork,
+            rpcUrls: [...new Set([...relayerNetwork.rpcUrls, ...storedNetwork.rpcUrls])]
+          }
+        }
+      })
+
+      // Step 3: Ensure predefined networks are marked correctly and handle special cases
+      let predefinedNetworkIds = Object.keys(updatedNetworks)
+
+      if (!predefinedNetworkIds.length) {
+        predefinedNetworkIds = predefinedNetworks.map((network) => network.chainId.toString())
+      }
+
+      Object.keys(updatedNetworks).forEach((chainId: string) => {
+        const network = updatedNetworks[chainId]
+
+        // If a predefined network is removed by the relayer, mark it as custom
+        // and remove the predefined flag
+        // Update the hasRelayer flag to false just in case
+        if (!predefinedNetworkIds.includes(network.chainId.toString()) && network.predefined) {
+          updatedNetworks[chainId] = { ...network, predefined: false, hasRelayer: false }
+        }
+
+        // Special case: Set the platformId for Odyssey chain
+        if (network.chainId === ODYSSEY_CHAIN_ID) {
+          updatedNetworks[chainId] = { ...network, platformId: 'ethereum' }
+        }
+      })
+    } catch (e: any) {
+      console.error('Failed to fetch networks from the Relayer', e)
+    }
+
+    return updatedNetworks
+  }
+
+  /**
+   * Updates network features asynchronously if needed.
+   */
+  #updateNetworkFeatures(finalNetworks: { [key: string]: Network }) {
+    Object.values(finalNetworks).forEach((network) => {
+      if (network.isSAEnabled) return
+
+      if (
+        network.lastUpdatedNetworkInfo &&
+        Date.now() - network.lastUpdatedNetworkInfo <= 24 * 60 * 60 * 1000
+      )
+        return
+
+      // eslint-disable-next-line @typescript-eslint/no-floating-promises
+      getNetworkInfo(
+        this.#fetch,
+        network.selectedRpcUrl,
+        network.chainId,
+        async (info) => {
```

Comment:
The getNetworkInfo function is called within a loop, and since it is asynchronous, it can potentially cause issues related to closures and scope.

I'm not sure if this is problematic, but please double-check 🙏 The best practice with these is wrapping all those in a Promise.all I think.